src/plugins/dnf/subscription_manager.py

```diff
@@ -206,6 +214,23 @@ def _warn_or_give_usage_message():
                     msg = not_registered_warning
             elif len(ent_dir.list_valid()) == 0 and not is_simple_content_access(identity=identity):
                 msg = no_subs_warning
+            else:
+                # Try to read release version ONLY from cache document.
+                # When cache document does not exist, then do not try to get this information
+                # from candlepin server and slow down DNF plugin!
+                release_cache: ReleaseStatusCache = inj.require(inj.RELEASE_STATUS_CACHE)
+                if release_cache.exists():
+                    release_version: Optional[str] = None
+                    cp_provider: CPProvider = inj.require(inj.CP_PROVIDER)
+                    cp = cp_provider.get_consumer_auth_cp()
+                    release_version_dict: Optional[dict] = release_cache.read_status(cp, identity.uuid)
```

Comment:
I think these lines can be simplified a bit by using `release_cache.read_cache_only()`, which should get the cached value only if it is available (and return `None` otherwise); I tested it locally, and it seems to work fine